src/crate_metadata.rs

```diff
@@ -41,7 +41,8 @@ pub struct CrateMetadata {
 impl CrateMetadata {
     /// Parses the contract manifest and returns relevant metadata.
     pub fn collect(manifest_path: &ManifestPath) -> Result<Self> {
-        let (metadata, root_package) = get_cargo_metadata(manifest_path)?;
+        let (mut metadata, root_package) = get_cargo_metadata(manifest_path)?;
+        metadata.target_directory.push("ink");
```

Comment:
It would be better to introduce a new field directly on our metadata struct for this rather than modifying the original `cargo` metadata struct. Current usages would use that field (or a getter) instead of `&crate_metadata.cargo_meta.target_directory`